python/ml4ir/base/features/preprocessing.py

```diff
@@ -109,15 +110,22 @@ def split_and_pad_string(feature_tensor, split_char=",", max_length=20):
         could returns the padded tokens ['AAA', 'BBB', 'CCC', '', '']
     """
     tokens = tf.strings.split(feature_tensor, sep=split_char).to_tensor()
-    padded_tokens = tf.image.pad_to_bounding_box(tf.expand_dims(tokens[:, :max_length], axis=-1),
-                                              offset_height=0,
-                                              offset_width=0,
-                                              target_height=1,
-                                              target_width=max_length)
+    padded_tokens = tf.image.pad_to_bounding_box(
+        tf.expand_dims(tokens[:, :max_length], axis=-1),
+        offset_height=0,
+        offset_width=0,
+        target_height=1,
+        target_width=max_length,
+    )
     padded_tokens = tf.squeeze(padded_tokens, axis=-1)
     return padded_tokens
 
 
+@tf.function
+def signed_log(feature_tensor, shift=1.0):
+    """Signed log"""
```

Comment:
We need to add extensive doc.
Mostly about why and what are the advantage, + typical usage for signed log in our use-case.

Reply:
I added the documentation. I don't think I meant to use signed_log. Reverting the name to unsigned_log